api-server/rhelai-install.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+# -*- indent-tabs-mode: nil; tab-width: 2; sh-indentation: 2; -*-
+
+# Install script for the API server
+## dependencies: go, git
+## required ENV vars: TAXONOMY_PATH, BASE_DIR
+
+set -x
+set -e
+set -o pipefail
+
+### installations
+
+rpm-ostree install gcc glibc-devel
+systemctl reboot
+export CGO_ENABLED=1
+
+### Validations
+
+if [ -z $(command -v git) ]; then
+    echo "please make sure \`git\` is installed."
+    exit 1 
+fi
+
+if [ -z $(command -v go) ]; then
+    echo "please make sure \`go\` is installed."
+    exit 1 
+fi
+
+if [ -z "$TAXONOMY_PATH"]; then
+    echo "please set the taxonomy-path as env variable \$TAXONOMY_PATH"
+    exit 1
+else
+    if [ ! -d "$TAXONOMY_PATH"]; then
+        echo "\$TAXONOMY_PATH was set as $TAXONOMY_PATH, but path does not exist."
```

Comment:
Let's set the default taxonomy to the default install. `$HOME/.local/share/instructlab/taxonomy/`. If that doesn't exist then prompt the user etc. We can set it in the api-server as a default as well as needed for simplicity.